irohad/main/application.cpp

```diff
@@ -100,14 +100,14 @@ void Irohad::dropStorage() {
  * Initializing iroha daemon storage
  */
 void Irohad::initStorage() {
-  auto factory =
+  common_objects_factory_ =
       std::make_shared<shared_model::proto::ProtoCommonObjectsFactory<
           shared_model::validation::FieldValidator>>();
   auto block_converter =
       std::make_shared<shared_model::proto::ProtoBlockJsonConverter>();
   auto storageResult = StorageImpl::create(block_store_dir_,
                                            pg_conn_,
-                                           std::move(factory),
+                                           common_objects_factory_,
```

Comment:
please keep std::move

Reply:
This shared pointer is reused in several components, so it cannot be moved.